src/compiler/types.ts

```diff
@@ -1511,6 +1517,32 @@ namespace ts {
         typeExpression: JSDocTypeExpression;
     }
 
+    // @kind(SyntaxKind.JSDocTypedefTag)
+    export interface JSDocTypedefTag extends JSDocTag, Declaration, JSDocParentTag {
+        name: Identifier;
+        typeExpression?: JSDocTypeExpression;
+        type: JSDocType;
+    }
+
+    export interface JSDocParentTag extends JSDocTag {
+        jsDocPropertyTags?: NodeArray<JSDocPropertyTag>;
+        jsDocTypeTag?: JSDocTypeTag;
+    }
+
+    // @kind(SyntaxKind.JSDocPropertyTag)
+    export interface JSDocPropertyTag extends JSDocTag, TypeElement {
+        name: Identifier;
+        typeExpression: JSDocTypeExpression;
+        // Add a "type" property here to make the interface compatible 
+        // with the "VariableLikeDeclaration" definition

```

Comment:
Could you add why you need to make it compatible with "VariableLikeDeclaration"?